src/Core/Billing/Models/MaskedPaymentMethodDTO.cs

```diff
@@ -0,0 +1,160 @@
+﻿using Bit.Core.Billing.Extensions;
+using Bit.Core.Enums;
+
+namespace Bit.Core.Billing.Models;
+
+public record MaskedPaymentMethodDTO(
+    PaymentMethodType Type,
+    string Description,
+    bool NeedsVerification)
+{
+    public static MaskedPaymentMethodDTO From(Stripe.Customer customer)
+    {
+        var defaultPaymentMethod = customer.InvoiceSettings?.DefaultPaymentMethod;
+
+        if (defaultPaymentMethod == null)
+        {
+            return customer.DefaultSource != null ? FromStripeLegacyPaymentSource(customer.DefaultSource) : null;
+        }
+
+        return defaultPaymentMethod.Type switch
+        {
+            "card" => FromStripeCardPaymentMethod(defaultPaymentMethod.Card),
+            "us_bank_account" => FromStripeBankAccountPaymentMethod(defaultPaymentMethod.UsBankAccount),
+            _ => null
+        };
+    }
+
+    public static MaskedPaymentMethodDTO From(Stripe.SetupIntent setupIntent)
+    {
+        if (!setupIntent.IsUnverifiedBankAccount())
+        {
+            return null;
+        }
+
+        var bankAccount = setupIntent.PaymentMethod.UsBankAccount;
+
+        var description = $"{bankAccount.BankName}, *{bankAccount.Last4}";
+
+        return new MaskedPaymentMethodDTO(
+            PaymentMethodType.BankAccount,
+            description,
+            true);
+    }
+
+    public static MaskedPaymentMethodDTO From(Braintree.Customer customer)
+    {
+        var defaultPaymentMethod = customer.DefaultPaymentMethod;
+
+        if (defaultPaymentMethod == null)
+        {
+            return null;
+        }
+
+        switch (defaultPaymentMethod)
+        {
+            case Braintree.PayPalAccount payPalAccount:
+                {
+                    return new MaskedPaymentMethodDTO(
+                        PaymentMethodType.PayPal,
+                        payPalAccount.Email,
+                        false);
+                }
+            case Braintree.CreditCard creditCard:
+                {
+                    var formattedExpirationMonth =
```

Comment:
⛏️ Given a `string`, you can pad the month with zeros to a width of two with `creditCard.ExpirationMonth.PadLeft(2, '0')`

Reply:
Updated in https://github.com/bitwarden/server/pull/4140/commits/1b9ac615459d3721c218fa4fb84446386f30d973